apps/src/lib/levelbuilder/lesson-editor/ObjectivesEditor.jsx

```diff
@@ -0,0 +1,209 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import color from '@cdo/apps/util/color';
+
+const styles = {
+  oddRow: {
+    backgroundColor: color.lightest_gray
+  },
+  actionButtons: {
+    display: 'flex',
+    justifyContent: 'space-evenly',
+    backgroundColor: 'white',
+    height: 30
+  },
+  remove: {
+    fontSize: 14,
+    color: 'white',
+    background: color.dark_red,
+    cursor: 'pointer',
+    textAlign: 'center',
+    minWidth: '50%',
+    lineHeight: '30px'
+  },
+  edit: {
+    fontSize: 14,
+    color: 'white',
+    background: color.default_blue,
+    cursor: 'pointer',
+    textAlign: 'center',
+    minWidth: '50%',
+    lineHeight: '30px'
+  },
+  save: {
+    fontSize: 14,
+    color: 'white',
+    background: color.green,
+    cursor: 'pointer',
+    textAlign: 'center',
+    minWidth: '50%',
+    lineHeight: '30px'
+  },
+  addButton: {
+    background: color.cyan,
+    borderRadius: 3,
+    color: color.white,
+    fontSize: 14,
+    padding: 7,
+    textAlign: 'center',
+    marginTop: 10,
+    marginLeft: 0
+  }
+};
+
+export default class ObjectivesEditor extends Component {
+  static propTypes = {
+    objectives: PropTypes.arrayOf(PropTypes.object)
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      objectives: props.objectives || [],
+      objectiveInput: '',
+      currentlyEditingIndex: null,
+      nextObjectiveKey: 1
```

Comment:
does this need to start at prop.objectives.length possibly?

Reply:
I don't think it matters, because the "keys" are constructed differently between persisted and not-yet-persisted objectives. 

I should probably do some verification that no other objective has the same key though.